x-pack/platform/plugins/shared/dataset_quality/server/routes/data_streams/update_failure_store/index.ts

```diff
@@ -30,7 +30,7 @@ export async function updateFailureStore({
           enabled: failureStoreEnabled,
           lifecycle: {
             data_retention: customRetentionPeriod,
-            ...(isServerless ? {} : { enabled: !!customRetentionPeriod }),
+            ...(isServerless ? {} : { enabled: failureStoreEnabled }),
```

Comment:
will this request update partially what we have in `lifecycle`?
Meaning that it will keep the other values
```
...
effective_retention: value,
retention_determined_by: value,
...
```
?
I think we are missing the update of those values, from what I see `effective_retention` should be equal to the value we set as `data_retention` (if we set a customRetentionPeriod) or cluster default (guessing here, if we don't set a customRetentionPeriod). Then `retention_determined_by` should be `default_failures_retention` or `data_stream_configuration` accordingly.

Reply:
ahhhh, nice! thanks